tests/test_schema_errors/test_strings.py

```diff
@@ -0,0 +1,110 @@
+import pytest
+
+from openapi_tester import DocumentationError, SchemaTester
+
+example_schema_string = {"type": "string", "minLength": 3, "maxLength": 5}
+example_string = "str"
+
+tester = SchemaTester()
+
+
+# region: string
+
+
+def test_nothing_wrong():
+    """ This should always pass """
+    tester.test_schema_section(example_schema_string, example_string, "")
+
+
+def test_min_length_violated():
+    """ Not adhering to minlength limitations should raise an error """
+    with pytest.raises(
+        DocumentationError,
+        match="Mismatched content. Response string violates the minimum string " "length defined in the schema",
+    ):
+        tester.test_schema_section(example_schema_string, "a" * 2, "")
+
+
+def test_max_length_violated():
+    """ Not adhering to maxlength limitations should raise an error """
+    with pytest.raises(
+        DocumentationError,
+        match="Mismatched content. Response string violates the maximum string " "length defined in the schema",
+    ):
+        tester.test_schema_section(example_schema_string, "a" * 6, "")
+
+
+def test_null():
+    """ A null value should always raise an error """
+    with pytest.raises(DocumentationError, match="Mismatched types, expected str but received NoneType"):
+        tester.test_schema_section(example_schema_string, None, reference="")
+
+
+def test_nullable():
+    """ A null value is allowed when the array is nullable """
+    schema = {"type": "string", "nullable": True}
+    tester.test_schema_section(schema, None, reference="")
+
+
+def test_wrong_type():
+    """ Type mismatches should raise errors """
+    with pytest.raises(DocumentationError, match="expected str but received dict."):
+        tester.test_schema_section(example_schema_string, {}, "")
+    with pytest.raises(DocumentationError, match="expected str but received list."):
+        tester.test_schema_section(example_schema_string, [], "")
+    with pytest.raises(DocumentationError, match="expected str but received int."):
+        tester.test_schema_section(example_schema_string, 1, "")
+    with pytest.raises(DocumentationError, match="expected str but received float."):
+        tester.test_schema_section(example_schema_string, 1.1, "")
+    with pytest.raises(DocumentationError, match="expected str but received tuple."):
+        tester.test_schema_section(example_schema_string, ("test",), "")
+
+
+# endregion: string
+
+# region: special formats
+
+
+def test_good_date():
+    tester.test_schema_section({"type": "string", "format": "date"}, "2040-01-01", "")
+
+
+def test_bad_date():
+    with pytest.raises(DocumentationError, match="expected a value with the format date but received 01-31-2019"):
```

Comment:
we can handle a lot of different date formats. This wont cut it. Put a really invalid date value.

Reply:
According to this https://swagger.io/docs/specification/data-models/data-types/#string we should **only** support [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) 

I purposefully put in a valid datetime format that just doesn't match ISO8601.

Reply:
Fair enough, BUT - this is a question we should consider: are we trying to follow the openapi specification to the letter? I think we should add a configurable "strict" flag, which dictates this behaviour. As a user of the library - i might decide to serialize/deserialize dates in different formats for the consumption of FE clients (e.g. a react based website for example). In that case what matters is that the date or datetime is a valid date / datetime string, within the Django eco-system. This is why I am using the django parsers for validation here.

Reply:
A strict flag sounds good. Perhaps we can add a separate issue for this, and handle it outside this PR?